src/Compilers/Core/Portable/CodeGen/SwitchIntegralJumpTableEmitter.cs

```diff
@@ -466,7 +466,20 @@ private void EmitRangeCheckedBranch(ConstantValue startConstant, ConstantValue e
             }
             else
             {
-                _builder.EmitIntConstant(endConstant.Int32Value - startConstant.Int32Value);
+                int Int32Value(ConstantValue value)
```

Comment:
Ah, I see the literals must have signed bits, which will force them to be 64bit and thus there will be no sign or zero extending. 
Basically the case is only relevant for switch types shorter than `uint`